pkg/replica/backup.go

```diff
@@ -154,24 +158,33 @@ type BackupStatus struct {
 	BackupURL     string
 	State         ProgressState
 	IsIncremental bool
+	IsOpened      bool
 }
 
-func NewBackup(backingFile *backingfile.BackingFile) *BackupStatus {
+func NewBackup(name, volumeID, snapID string, backingFile *backingfile.BackingFile) *BackupStatus {
+	backupName := name
+	if backupName == "" {
+		backupName = butil.GenerateName("backup")
+	}
 	return &BackupStatus{
+		Name:        backupName,
 		backingFile: backingFile,
-		State:       ProgressStateInProgress,
+		State:       ProgressStateStarting,
+		volumeID:    volumeID,
+		SnapshotID:  diskutil.GenerateSnapshotDiskName(snapID),
 	}
 }
 
-func (rb *BackupStatus) UpdateBackupStatus(snapID, volumeID string, progress int, url string, errString string) error {
+func (rb *BackupStatus) UpdateBackupStatus(snapID, volumeID string, state string, progress int, url string, errString string) error {
```

Comment:
This function looks weird now. It receives an input state but sometimes it determines the state by itself...

Reply:
Indeed, so I would suggest at least having a comment to explain this progress first-respected behavior.

Reply:
Add a comment as @innobead's suggestion.